src/app/stores/window-store.ts

```diff
@@ -66,16 +81,36 @@ export class WindowStore {
     if (hideOnCapture) {
       const currentWindows = this.getWindowStore();
       currentWindows.windows.forEach((currentWindow) => {
-        if (!currentWindow.minimized) {
-          getWindowByName(currentWindow.id || '')?.show();
-        }
+        const window = getWindowByName(currentWindow.id || '');
+        if (window) {
+          if (!currentWindow.minimized) {
+            window.show();
+          }
 
-        if (currentWindow.focused) {
-          getWindowByName(currentWindow.id || '')?.focus();
+          if (currentWindow.isFullScreen) {
+            window.setFullScreen(true);
+          }
+
+          if (currentWindow.focused) {
+            window.focus();
+          }
         }
       });
 
       this.destroyWindowStore();
     }
   };
+
+  private hideFullscreenWindow = (window: BrowserWindow) => {
+    window.once('leave-full-screen', () => {
+      if (isMac) {
+        window.hide();
+      } else {
+        setTimeout(() => {
```

Comment:
LGTM, can be merged.
Except one thing that the timeout here, is Windows having some sort of timing issue? while mac doesn't have it